File: page-objects/js-protractor/complete/pages/home-page.js
var SearchResultsPage = require('./search-results-page.js');

HomePage = function HomePage() {
    this.searchField = element(by.id('searchContent'));
};

HomePage.prototype.visit = function () {
    browser.get('https://www.zalando.de/mosaic-mosaic-catalog');
};

HomePage.prototype.search = function (searchText) {
    this.searchField.sendKeys(searchText);
    this.searchField.submit();
    return new SearchResultsPage();
};

module.exports = HomePage;

File: page-objects/java-fluentlenium/complete/src/main/java/pages/HomePage.java
package pages;

import org.fluentlenium.core.FluentPage;
import org.fluentlenium.core.annotation.Page;
import org.openqa.selenium.By;

import static org.assertj.core.api.Assertions.assertThat;

public class HomePage extends FluentPage {

    @Page
    private SearchResultsPage searchResultsPage;

    @Override
    public String getUrl() {
        return "https://www.zalando.de/";
    }

    @Override
    public void isAt() {
        assertThat(window().title().toLowerCase()).contains("zalando");
    }

    public SearchResultsPage search(String searchText) {
        find(By.id("searchContent")).write(searchText).submit();
        return searchResultsPage;
    }

}
